### ISSUE TITLE
[FEATURE] - arrange bookmarks

### ISSUE THREAD

**Issue by whnmgdsdb** [alefragnani/vscode-bookmarks#655]
new feature needed:
1.The order of bookmarks need based on my own preferences (currently by source code line number).The order of the bookmark is very important to me.
2.Drag & drop bookmarks to sort and arrange to myliking.
3.Add group function to store bookmarks.Group is like a folder to store different kinds of bookmarks

![image](https://github.com/alefragnani/vscode-bookmarks/assets/85162742/417e4a2f-dccc-4dda-895d-2b9b8d762f01)

**Comment by whnmgdsdb** [comment]
if recerive the issue message please reply me.

**Comment by grantyuc** [comment]
Upvote!

**Comment by ckx** [comment]
I would highly appreciate all of these features as well.

**Comment by twcctop** [comment]
+1, need this features

**Comment by tommyme** [comment]
+1, like it.

**Comment by tommyme** [comment]
In fact, vscode now support drag and drop api in TreeView [official example](https://github.com/microsoft/vscode-extension-samples/blob/main/tree-view-sample/src/testViewDragAndDrop.ts).
R u willing to add the functionality to do that? Or you can open source the CORE code so that other developers can do Pull Requests.

**Comment by tommyme** [comment]
@alefragnani plase consider the feature 😊

**Comment by Adgamby** [comment]
The extension would be perfect with this feature

**Comment by tommyme** [comment]
Just try my bookmark plugin, you can arrange bookmark order by naming, and you can create nested group
just search bookmarkx in vscode plugin marketplace.

**Comment by Juriy** [comment]
Would be nice to have this feature. I am working now on a complex assessment and analysis project, where the team has to explore a vast code base and code flows - having folders and ordering would speed up the process a lot.

**Comment by meijialun** [comment]
I have been asking order feature quite long ago. Sad.

**Comment by man8266** [comment]
> Just try my bookmark plugin, you can arrange bookmark order by naming, and you can create nested group just search bookmarkx in vscode plugin marketplace.

Hi Tommyme, I have tried your "bookmark X" for weeks, it can sorted labels, which other bookmark don't have. But with "bookmark X", it very often need too much time to list all bookmarks, or show current bookmark on certain lines, often take about 40s-60s, which make me very annoyed when I am busy in on investigate some code. Can you improve it? Otherwise, your "bookmark X" is the best by now.

**Comment by man8266** [comment]
I think it is very important for me to arrange bookmark order by naming, when I added a lot of bookmarks for reading a big project.

**Comment by tommyme** [comment]
okey, I'll work on it later.
I don't understand what do you mean by `show current bookmark on certain lines`
bookmark x plugin can do `reveal bookmark in current line`, which need to be called with command palette

more issues can be discussed in bookmark X repo

**Comment by man8266** [comment]
> I don't understand what do you mean by `show current bookmark on certain lines`

Yes. "bookmark_x.revealBookmark" is exactly what I want to do (I just add a shortcut ctlr+alt+N to do that). 
But sometimes (maybe happen 5-6 times when I am investigating the code for half hour), even just above ctlr+alt+N command need to be finished in 30-60s.

But I don't 100% sure this problem is caused by "bookmark X" extension itself, or something else affect "bootmark X". It can happen on my Windows, Ubuntu machines. Not matter investigating code which located in local machine, or by SSH to anther machine.

**Comment by tommyme** [comment]
> matter 

ok,thanks alot, I'll try to reproduce it on my machine.

**Comment by man8266** [comment]
> > matter
> 
> ok,thanks alot, I'll try to reproduce it on my machine.

Hi Tommyme. For last 2 days, I enabled View->Appearance->Secondary Side Bar. So I let BookMark X always displayed in this Secondary Side Bar. I did not see the issue (when add/delete bookmarks, or show current bookmark in current code line) I reported several days ago.

**Comment by hartmut27** [comment]
in the meantime, as a workaround, goto setting "bookmarks.saveBookmarksInProject", enable this, you see your individual file `bookmarks.json` in your project (`yourprojectxyz/.vscode/bookmarks.json`) , edit this file, change the ordering of your entries as you wish, save it, restart vscode, et voilà, you've got your ordering of bookmarks.
Love this plugin! helps much.

**Comment by MaGeBZz** [comment]
big upvote too from me but big shoutouts to @hartmut27 with the temporary fix, haven't thought about this but you saved my workflow with it, thanks :D

**Comment by AgainPsychoX** [comment]
Duplicate of #378, but I guess that was closed with introduction of sorting, so this should stay.

Either way: Bump! Would love to see this feature :)
